### PR TITLE
use powershell_exec instead of powershell_out in windows_feature_powershell

### PR DESCRIPTION
**Pull request by rishichawda**
Signed-off-by: rishichawda <rishichawda@users.noreply.github.com>

## Description

`powershell_exec` has better performance compared to `powershell_out`. This PR updates the `windows_feature_powershell` resource to use `powershell_exec`

## Related Issue

#10927 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tas50** [comment]
Can we get an integration test added here for that resource using the smallest feature you can find on windows: https://github.com/chef/chef/blob/main/kitchen-tests/cookbooks/end_to_end/recipes/windows.rb

**Comment by rishichawda** [comment]
@tas50 I've add the test but this PR needs to wait for https://github.com/chef/chef/pull/11984#issuecomment-911917172

File: kitchen-tests/kitchen.vagrant.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+driver:
```
Comment (tas50) [inline kitchen-tests/kitchen.vagrant.yml:1]: This kitchen config isn't going to actually test the current content of chef/chef. It'll just pull the latest stable build. You probably want to remove it from this PR.

Comment (rishichawda) [reply kitchen-tests/kitchen.vagrant.yml]: Done.

**Comment by lamont-granquist** [comment]
possibly relevant test failures here as well?

**Comment by rishichawda** [comment]
@lamont-granquist Will rebase this once #11984 passes. It should work then.

**Review by lamont-granquist — APPROVED** [review]

**Comment by lamont-granquist** [comment]
This is good providing the tests go green.

**Comment by rishichawda** [comment]
Rebased the branch on main. GitHub workflows are passing now.

File: lib/chef/resource/windows_feature_powershell.rb [hunk 2]
```diff
@@ -215,7 +215,7 @@ def reload_cached_powershell_data
         # fetch the list of available feature names and state in JSON and parse the JSON
         def parsed_feature_list
           # Grab raw feature information from WindowsFeature
-          raw_list_of_features = powershell_out!("Get-WindowsFeature | Select-Object -Property Name,InstallState | ConvertTo-Json -Compress", timeout: new_resource.timeout).stdout
+          raw_list_of_features = powershell_exec!("Get-WindowsFeature | Select-Object -Property Name,InstallState | ConvertTo-Json -Compress", timeout: new_resource.timeout).result
```
Comment (tas50) [inline lib/chef/resource/windows_feature_powershell.rb:218]: If we're using powershell_exec we don't need to convert to json. The data should be returned in JSON to start with.

Comment (rishichawda) [reply lib/chef/resource/windows_feature_powershell.rb]: done!

**Pull request merged**